xbmc/platform/linux/CPUInfoLinux.cpp

```diff
@@ -371,11 +379,18 @@ bool CCPUInfoLinux::GetTemperature(CTemperature& temperature)
   if (m_tempPath.empty())
     return false;
 
-  CSysfsPath path{m_tempPath};
-  double value = path.Get<double>() / 1000.0;
+  try
+  {
+    CSysfsPath path{m_tempPath};
+    double value = path.Get<double>() / 1000.0;
```

Comment:
Does this throw or are you just being defensive? Maybe it would be better to build the safety into `CSysfsPath`?

Reply:
I'm just being defensive here.

Moving this safety to `CSysfsPath` sounds like a good idea. Are you thinking about catching exceptions in `CSysfsPath::Get<>()` and changing the return value of that method to a `std::optional` to distinguish failure/success?

Reply:
Yea that would be ok with me.